components/sync15/src/status.rs

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::{Error, ErrorKind, StorageHttpError};
+
+/// The general status of sync - should probably be moved to the "sync manager"
+/// once we have one!
+#[derive(Debug, PartialEq)]
+pub enum ServiceStatus {
+    /// Everything is fine.
+    Ok,
```

Comment:
Part of me feels like this probably should just be an enum of the error type, and we should be using `Result<_, ServiceStatus>` or `Option<ServiceStatus>` elsewhere, but it's probably fine for now.

Reply:
There's now a `SyncResult` - no `Result` or `Option`s in sight (well, kinda - lmk what you think of the new version)